src/tsickle.ts

```diff
@@ -630,6 +638,21 @@ class Annotator extends ClosureRewriter {
     return toArray(reexports.keys());
   }
 
+  private writeModuleSpecifier(moduleSpecifier: ts.Expression) {
+    let moduleId = (moduleSpecifier as any).text;
+    if (this.tsOpts && this.host) {
+      const resolved = ts.resolveModuleName(moduleId, this.file.fileName, this.tsOpts, this.host);
+      if (resolved && resolved.resolvedModule) {
+        const resolvedModule = resolved.resolvedModule.resolvedFileName.replace(/(\.d)?\.ts$/, '');
+
+        if (resolvedModule.match(/\/index$/) && !moduleId.match(/\/index$/)) {
+          moduleId += '/index';
```

Comment:
good point, I need a test for that, Mohamed says that is a workaround:
https://github.com/Microsoft/TypeScript/issues/12597#issuecomment-264254009